Orange/widgets/visualize/owmap.py

```diff
@@ -0,0 +1,837 @@
+            callback=lambda: self.map.set_marker_label(self.label_attr))
+        combo.setModel(self._label_model)
+        self._combo_shape = combo = gui.comboBox(
+            box, self, 'shape_attr',
+            orientation=Qt.Horizontal,
+            label='Shape:',
+            sendSelectedValue=True,
+            callback=lambda: self.map.set_marker_shape(self.shape_attr))
+        combo.setModel(self._shape_model)
+        self._combo_size = combo = gui.comboBox(
+            box, self, 'size_attr',
+            orientation=Qt.Horizontal,
+            label='Size:',
+            sendSelectedValue=True,
+            callback=lambda: self.map.set_marker_size(self.size_attr))
+        combo.setModel(self._size_model)
+
+        def _set_opacity():
+            map.set_marker_opacity(self.opacity)
+
+        def _set_zoom():
+            map.set_marker_size_coefficient(self.zoom)
+
+        def _set_jittering():
+            map.set_jittering(self.jittering)
+
+        def _set_clustering():
+            map.set_clustering(self.cluster_points)
+
+        self._opacity_slider = gui.hSlider(
+            box, self, 'opacity', None, 1, 100, 5,
+            label='Opacity:', labelFormat=' %d%%',
+            callback=_set_opacity)
+        self._zoom_slider = gui.valueSlider(
+            box, self, 'zoom', None, values=(20, 50, 100, 200, 500, 1000),
+            label='Symbol size:', labelFormat=' %d%%',
+            callback=_set_zoom)
+        self._jittering = gui.valueSlider(
+            box, self, 'jittering', label='Jittering:', values=(0, .5, 1, 2, 5),
+            labelFormat=' %.1f%%', ticks=True,
+            callback=_set_jittering)
+        self._clustering_check = gui.checkBox(
+            box, self, 'cluster_points', label='Cluster points',
+            callback=_set_clustering)
+
+        gui.rubber(self.controlArea)
+        gui.auto_commit(self.controlArea, self, 'autocommit', 'Send Selection')
+
+        QTimer.singleShot(0, _set_map_provider)
+        QTimer.singleShot(0, _toggle_legend)
+        QTimer.singleShot(0, _set_opacity)
+        QTimer.singleShot(0, _set_zoom)
+        QTimer.singleShot(0, _set_jittering)
+        QTimer.singleShot(0, _set_clustering)
+
+    autocommit = settings.Setting(True)
+
+    def __del__(self):
+        self.progressBarFinished(None)
+        self.map = None
+
+    def commit(self):
+        self.send('Selected Data', self.selection)
+        self.send(ANNOTATED_DATA_SIGNAL_NAME,
+                  create_annotated_table(self.data, self._indices))
+
+    def set_data(self, data):
+        self.data = data
+
+        self.closeContext()
+
+        if data is None:
+            return self.clear()
+
+        all_vars = list(chain(self.data.domain, self.data.domain.metas))
+        continuous_vars = [var for var in all_vars if var.is_continuous]
+        discrete_vars = [var for var in all_vars if var.is_discrete]
+        primitive_vars = [var for var in all_vars if var.is_primitive]
```

Comment:
This filtering doesn't seem to work — i.e. `primitive_vars` is the same as `all_vars` even when StringVariables are present. Isn't `is_primitive` a method rather than a property?